### PR TITLE
add pay in 4 apply now flow to dev page

### PR DESCRIPTION
**Pull request by bakedbean**
switch payin4 checkout experience based on url subdomain
Moved dispatching of cart actions from the product model to the cart hook.

### CODE REVIEW

**Comment by bakedbean** [comment]
for local dev you can add `payin4.localhost` to your /etc/hosts file to reproduce testing a pay in 4 merchant public key

File: src/components/store/pages/productShow.js [hunk 1]
```diff
@@ -53,7 +56,7 @@ export default () => {
             <div className="buy-it-now">
               <button
                 className="button is-medium is-fullwidth"
-                onClick={() => product.addToCart()}
+                onClick={() => removeProductFromCart(product)}
```
Comment (widox) [inline src/components/store/pages/productShow.js:59]: seems like you might want `addProductToCart` here eh?

File: src/lib/utils.js [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
 
 const publicKeys = {
-  development: 'thewebstaurantstoreinc_1a3fc138f571474b997815ab57febbfe',
+  development: 'creditkeydev_2822baea77774929979f1e5964dd18b6',
   staging: 'creditkeytest_6d8e5758033846b4995993dd74dda57c',
   production: 'creditkeysamplestore_2d3170af046347cdaf0171e06c08ca77'
 }
 
+const pi4PublicKeys = {
+  development: 'thewebstaurantstoreinc_dcf536bfabb748999f1bf57dda2630c6', // eben local
```
Comment (widox) [inline src/lib/utils.js:10]: it would be very useful to be able to define these keys in the `.env` files  😄

Comment (bakedbean) [reply src/lib/utils.js:10]: agreed

File: src/components/ApplyFlow.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ export default function ApplyFlow({ applyFlow, handleFlow }) {
           value={applyFlow}
           onChange={e => handleFlow(e.target.value)}
         >
+          <option>Pay in 4 Apply Now</option>
```
Comment (widox) [inline src/components/ApplyFlow.js:17]: nitpick: I guess this isn't always true if you're not on a pi4 subdomain

**Review by widox — APPROVED** [review]

**Pull request merged**